decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Decorator < Person
+  def initialize(person)
+    super(person.id, person.age, parent_permission: person.parent_permission, name: person.name)
+    @person = person
+  end
+
+  def correct_name
+    @person.correct_name
+  end
```

Comment:
- [ ] Please make sure that the method returns the result of the `correct_name` method of the `@nameable` ,as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Implement%20the%20correct_name%20method%20that%20returns%20the%20result%20of%20the%20correct_name%20method%20of%20the%20%40nameable.) requirement.
```ruby
@nameable.correct_name
```